src/ddmd/func.d

```diff
@@ -4213,11 +4213,15 @@ extern (C++) FuncDeclaration resolveFuncCall(Loc loc, Scope* sc, Dsymbol s,
         TypeFunction tf2 = m.nextf.type.toTypeFunction();
         const(char)* lastprms = parametersTypeToChars(tf1.parameters, tf1.varargs);
         const(char)* nextprms = parametersTypeToChars(tf2.parameters, tf2.varargs);
-        .error(loc, "%s.%s called with argument types %s matches both:\n%s:     %s%s\nand:\n%s:     %s%s",
+
+        const(char)* mod1 = MODtoChars(tf1.mod);
+        const(char)* mod2 = MODtoChars(tf2.mod);
+
+        .error(loc, "%s.%s called with argument types %s matches both:\n%s:     %s%s %s\nand:\n%s:     %s%s %s",
             s.parent.toPrettyChars(), s.ident.toChars(),
             fargsBuf.peekString(),
-            m.lastf.loc.toChars(), m.lastf.toPrettyChars(), lastprms,
-            m.nextf.loc.toChars(), m.nextf.toPrettyChars(), nextprms);
+            m.lastf.loc.toChars(), m.lastf.toPrettyChars(), lastprms, mod1,
+            m.nextf.loc.toChars(), m.nextf.toPrettyChars(), nextprms, mod2);
```

Comment:
@RazvanN7 there's no need for code duplication if you include the space in the string that get's passed to `%s`. It can be something along the lines of:

```D
const(char)* mod1 = MODtoChars(tf1.mod).prependSpace();
const(char)* mod2 = MODtoChars(tf2.mod).prependSpace();

.error(loc, "%s.%s called with argument types %s matches both:\n%s:     %s%s%s\nand:\n%s:     %s%s%s",
    s.parent.toPrettyChars(), s.ident.toChars(),
    fargsBuf.peekString(),
    m.lastf.loc.toChars(), m.lastf.toPrettyChars(), lastprms, mod1,
    m.nextf.loc.toChars(), m.nextf.toPrettyChars(), nextprms, mod2);
```

Where `prependSpace` can be:
```D
const(char)* prependSpace(const char* str)
{
    import core.stdc.string : strlen;
    
    if (!str || !*str) return "";
    return (" " ~ str[0 .. strlen(str)] ~ "\0").ptr;
}
```

or this if D's built-in string concatenation can't be used in ddmd:

```D
const(char)* prependSpace(const char* str)
{
    import core.stdc.string : strlen;
    
    if (!str || !*str) return "";

    size_t len = strlen(str);
    auto res = new char[](1 + len + 1);
    res[0] = ' ';
    res[1 .. $ - 1] = str[0 .. len];
    res[$-1] = '\0';
    return res.ptr;
}
```

Or hopefully something cleaner, that I haven't though about.